crates/bevy_pbr/src/render/pbr_fragment.wgsl

```diff
@@ -0,0 +1,185 @@
+
+    // NOTE: Unlit bit not set means == 0 is true, so the true case is if lit
+    if ((pbr_bindings::material.flags & pbr_types::STANDARD_MATERIAL_FLAGS_UNLIT_BIT) == 0u) {
+
+        pbr_input.material.reflectance = pbr_bindings::material.reflectance;
+        pbr_input.material.alpha_cutoff = pbr_bindings::material.alpha_cutoff;
+
+        // TODO use .a for exposure compensation in HDR
+        var emissive: vec4<f32> = pbr_bindings::material.emissive;
+#ifdef VERTEX_UVS
+        if ((pbr_bindings::material.flags & pbr_types::STANDARD_MATERIAL_FLAGS_EMISSIVE_TEXTURE_BIT) != 0u) {
+            emissive = vec4<f32>(emissive.rgb * textureSampleBias(pbr_bindings::emissive_texture, pbr_bindings::emissive_sampler, uv, view.mip_bias).rgb, 1.0);
+        }
+#endif
+        pbr_input.material.emissive = emissive;
+
+        var metallic: f32 = pbr_bindings::material.metallic;
+        var perceptual_roughness: f32 = pbr_bindings::material.perceptual_roughness;
+#ifdef VERTEX_UVS
+        if ((pbr_bindings::material.flags & pbr_types::STANDARD_MATERIAL_FLAGS_METALLIC_ROUGHNESS_TEXTURE_BIT) != 0u) {
+            let metallic_roughness = textureSampleBias(pbr_bindings::metallic_roughness_texture, pbr_bindings::metallic_roughness_sampler, uv, view.mip_bias);
+            // Sampling from GLTF standard channels for now
+            metallic *= metallic_roughness.b;
+            perceptual_roughness *= metallic_roughness.g;
+        }
+#endif
+        pbr_input.material.metallic = metallic;
+        pbr_input.material.perceptual_roughness = perceptual_roughness;
+
+        // TODO: Split into diffuse/specular occlusion?
+        var occlusion: vec3<f32> = vec3(1.0);
+#ifdef VERTEX_UVS
+        if ((pbr_bindings::material.flags & pbr_types::STANDARD_MATERIAL_FLAGS_OCCLUSION_TEXTURE_BIT) != 0u) {
+            occlusion = vec3(textureSampleBias(pbr_bindings::occlusion_texture, pbr_bindings::occlusion_sampler, uv, view.mip_bias).r);
+        }
+#endif
+#ifdef SCREEN_SPACE_AMBIENT_OCCLUSION
+        let ssao = textureLoad(screen_space_ambient_occlusion_texture, vec2<i32>(in.position.xy), 0i).r;
+        let ssao_multibounce = gtao_multibounce(ssao, pbr_input.material.base_color.rgb);
+        occlusion = min(occlusion, ssao_multibounce);
+#endif
+        pbr_input.occlusion = occlusion;
+
+#ifdef LOAD_PREPASS_NORMALS
+        pbr_input.N = bevy_pbr::prepass_utils::prepass_normal(in.position, 0u);
+#else
+        pbr_input.N = pbr_functions::apply_normal_mapping(
+            pbr_bindings::material.flags,
+            pbr_input.world_normal,
+#ifdef VERTEX_TANGENTS
+#ifdef STANDARDMATERIAL_NORMAL_MAP
+            in.world_tangent,
+#endif
+#endif
+#ifdef VERTEX_UVS
+            uv,
+#endif
+            view.mip_bias,
+        );
+#endif
+
+        pbr_input.occlusion = occlusion;
+
+        pbr_input.flags = mesh[in.instance_index].flags;
+    }
+
+    return pbr_input;
+}
+
+fn in_shader_post_processing(
```

Comment:
I don't love using the term post processing, as it has overlap with things like bloom, taa, etc. Maybe call it `in_shader_post_lighting_adjustment` or something?

Reply:
that's why it's "in-shader" post processing. given that 2 of these things are normally done in actual post-processing i think the name's fair.

Reply:
I'm worried users will think that means bloom is done in the shader here, for instance.

Reply:
added clarifications in the example and the base pbr shader, as well as on the function

Reply:
I didn't really like the `in_shader` part when reading it, but couldn't come up with anything I liked better at the time. I was thinking something like `main_pass_...` so maybe a combination like `main_pass_post_lighting_processing`?

Reply:
done